### PR TITLE
Adding oceanoptics spectrometer hardware file

### PR DESCRIPTION
**Pull request by jochen-scheuer**
## Description

Adding a hardware file for the ocean optics spectrometer

## Motivation and Context

New hardware.

## How Has This Been Tested?

Tested by using the spectrometer QE-Pro on Windows10 with two different computers.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have documented my changes in the changelog (`documentation/changelog.md`)
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [x] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [x] All changed Jupyter notebooks have been stripped of their output cells.

### CODE REVIEW

File: hardware/spectrometer/oceanoptics_spectrometer.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        module.Class: 'spectrometer.oceanoptics_spectrometer.OceanOptics'
+        spectrometer_serial: 'QEP01583' #insert here the right serial number.
+
+    """
+    _serial = ConfigOption('spectrometer_serial', missing='warn')
+
+    def on_activate(self):
+        """ Activate module.
+        """
+        self.integration_time = 1e4
```
Comment (Neverhorst) [inline hardware/spectrometer/oceanoptics_spectrometer.py]: Hard-coding measurement parameters (especially in hardware modules) is problematic.
Isn't this initialization of the integration time unnecessary anyways? The device has a time set regardless of qudi.

File: hardware/spectrometer/oceanoptics_spectrometer.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        return specdata
+
+    def getExposure(self):
+        """ Get exposure.
+
+            @return float: exposure
+
+            Not implemented.
+        """
+        return self.integration_time_micros
```
Comment (Neverhorst) [inline hardware/spectrometer/oceanoptics_spectrometer.py]: Right after `__init__` this module instance has no attribute `integration_time_micros` and thus will raise `AttributeError`.
Always declare all instance variables in `__init__` first to avoid such a thing.

But I don't see why this variable is necessary? `integration_time_micros` seems to be stored in `spec` as well. So why not simply set and get this value in/from `spec` and avoid an additional copy which can possibly deviate from the true value set in the device?

File: hardware/spectrometer/oceanoptics_spectrometer.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+from core.module import Base, ConfigOption
+from interface.spectrometer_interface import SpectrometerInterface
+import numpy as np
+import seabreeze.spectrometers as sb
+
+
+
+class OceanOptics(Base, SpectrometerInterface):
+    """ Hardware module for reading spectra from the WinSpec32 spectrometer software.
```
Comment (Neverhorst) [inline hardware/spectrometer/oceanoptics_spectrometer.py]: I'm not sure but does the _Ocean Optics_ spectrometer use _WinSpec32_ or is this just a copy-paste leftover from another module?

File: hardware/spectrometer/oceanoptics_spectrometer.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Qudi is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Qudi. If not, see <http://www.gnu.org/licenses/>.
+
+Derived from the pyPL project (https://github.com/kaseyrussell/pyPL)
```
Comment (Neverhorst) [inline hardware/spectrometer/oceanoptics_spectrometer.py]: Is this really related to this module?

**Comment by kay-jahnke** [comment]
Can you please also add a changelog and perform the tests defined by the checkbox list.

File: hardware/spectrometer/oceanoptics_spectrometer.py (hunk 1 above)
Comment (jochen-scheuer) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: Actually, this is an important parameter and one has to be able to change it to work with the spectrometer. It should be possible to change this value in the gui. I changed it into a status variable such that it at least does not have to be set everytime.

File: hardware/spectrometer/oceanoptics_spectrometer.py (hunk 3 above)
Comment (jochen-scheuer) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: Was an error and is fixed, Thanks.

File: hardware/spectrometer/oceanoptics_spectrometer.py (hunk 4 above)
Comment (jochen-scheuer) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: Also a copy paste error, got fixed.

File: hardware/spectrometer/oceanoptics_spectrometer.py (hunk 2 above)
Comment (jochen-scheuer) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: fixed with StatusVar

File: hardware/spectrometer/oceanoptics_spectrometer.py (hunk 1 above)
Comment (Neverhorst) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: I was not suggesting to get rid of the possibility to change the exposure time.
I was just asking if you can read back this value from `spec`. If you can set the value with `self.spec.integration_time_micros(my_value)` there may also be a possibility to read `self.spec.integration_time_micros`. This would get rid of a possible error source and reduce redundancy by getting rid of the instance variable `self.integration_time`.
You would then still be able to set and get the integration time by calling `setExposure` and `getExposure`.

Comment (kay-jahnke) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: I looked into the code of the Seabreeze module: https://github.com/ap--/python-seabreeze/blob/master/seabreeze/spectrometers.py
Interestingly it does not have a way to query the integration time. Therefore I think Jochens implementation is valid.

Comment (Neverhorst) [reply hardware/spectrometer/oceanoptics_spectrometer.py]: Ok, cool, that's all I wanted to know. In that case the current solution is the best option.
